Yoast/wordpress-seo
# Decouple snippet data and initialization cleanup

## Context


* Snippet editor has more complex way of using the hidden fields. We want to make the yoast editor store as the source of truth and sync values to the hidden fields.

## Summary


This PR can be summarized in the following changelog entry:

* Remove the old sync methods and variables.

## Relevant technical choices:

* setDataFromSnippet is used as a sync function to the hidden fields for the snippet editor data. I left only the case for slug since it is not meta and the sync is done within the editor. 
* We set focus keyword in the initial state regardless of the analysis feature active or not. It doesn't make a difference.
* In the feature branch I added is_cornerstone to the initial state, so I'm removing the initializing and the syncing from the post data collector.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Without premium or other Yoast add-ons:
* Edit a post.
* Go to Google preview
* Change title, description and slug, try also with replacement variables
* Save and refresh, check it is saved
* Change focus keyphrase and is cornerstone and save.
* Refresh and check it is saved.

Repeat the tests for 
* The term editor
* The elementor editor 
* The classic editor

#### Test without metabox
- Comment out line 81 in `class-metabox.php`:
```php
// add_action( 'add_meta_boxes', [ $this, 'add_meta_box' ] );
```
* Edit a post.
* Go to Google preview
* Change title, description and slug, try also with replacement variables
* Save and refresh, check it is saved.

#### Relevant test scenarios


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:



## UI changes


## Other environments


## Documentation


## Quality assurance


## Innovation


Fixes https://github.com/Yoast/wordpress-seo/issues/21225
